tests/unit/processor/datetime_extractor/test_date_time_extractor.py

```diff
@@ -134,6 +132,83 @@ def test_an_event_extracted_datetime_and_local_utc_without_delta(self):
         }
         assert document == expected
 
+    def test_deletes_source_field(self):
+        document = {"@timestamp": "2019-07-30T14:37:42.861+00:00", "winlog": {"event_id": 123}}
+        rule = {
+            "filter": "@timestamp",
+            "datetime_extractor": {
+                "source_fields": ["@timestamp"],
+                "target_field": "split_@timestamp",
+                "delete_source_fields": True,
+            },
+            "description": "",
+        }
+        self._load_specific_rule(rule)
+        self.object.process(document)
+        expected = {
+            "winlog": {"event_id": 123},
+            "split_@timestamp": {
+                "year": 2019,
+                "month": 7,
+                "day": 30,
+                "hour": 14,
+                "minute": 37,
+                "second": 42,
+                "microsecond": 861000,
+                "weekday": "Tuesday",
+                "timezone": "UTC",
+            },
+        }
+        assert document == expected
+
+    def test_overwrite_target(self):
```

Comment:
This test doesn't run on my local machine. I think you have to add the following lines before the `self.object.process(document)` line, similar to the other tests. 

```
self.object._local_timezone = tzutc()
self.object._local_timezone_name = DatetimeExtractor._get_timezone_name(
    self.object._local_timezone
)
```

Reply:
done